pliers/extractors/text.py

```
@@ -453,98 +446,282 @@ def __init__(self,
+            the string, only the first one is masked. 
+        top_n (int): Specifies how many of the highest-probability tokens are
+            to be returned. Mutually exclusive with target and threshold.
+        target (str or list): Vocabulary token(s) for which probability is to 
+            be returned. Tokens defined in the vocabulary change across 
+            tokenizers. Mutually exclusive with top_n and threshold.
+        threshold (float): If defined, only values above this threshold will
+            be returned. Mutually exclusive with top_n and target.
+        return_softmax (bool): if True, returns probability scores instead of 
+            raw predictions scores.
+        return_metadata (bool): if True, returns masked word (if defined in the
+            tokenizer vocabulary) and its probability.
+        model_kwargs (dict): Named arguments for pretrained model.
+            See: https://huggingface.co/transformers/main_classes/model.html
+            and https://huggingface.co/transformers/model_doc/bert.html.
+        tokenizer_kwargs (dict): Named arguments for tokenizer.
+            See https://huggingface.co/transformers/main_classes/tokenizer.html.
+    '''
 
-        result_df = pd.DataFrame(df_dict)
-        result_df['object_id'] = range(result_df.shape[0])
+    _log_attributes = ('pretrained_model', 'framework', 'top_n', 'target', 'tokenizer_type', 'return_softmax')
 
-        return result_df
+    def __init__(self,
+                 pretrained_model='bert-base-uncased',
+                 tokenizer='bert-base-uncased',
+                 framework='pt',
+                 mask='[MASK]',
+                 top_n=100,
+                 threshold=None,
+                 target=None,
+                 return_softmax=False,
+                 return_true=False,
+                 model_kwargs=None,
+                 tokenizer_kwargs=None):
+        
+        super(BertLMExtractor, self).__init__(pretrained_model=pretrained_model, 
```

Here too you can probably move the superclass init to the end in order to fail faster in the event the validations fails (if you need to overwrite instance attributes set in the superclass, you can still put them after the superclass init).